Tests/NIOSSLTests/TLSConfigurationTest.swift

```diff
@@ -464,6 +525,220 @@ class TLSConfigurationTest: XCTestCase {
 
         try assertHandshakeSucceeded(withClientConfig: clientConfig, andServerConfig: serverConfig)
     }
+    
+    func testFullVerificationWithCANamesFromCertificate() throws {
+        // Custom certificates for TLS and client authentication.
+        let root = try NIOSSLCertificate(bytes: .init(customCARoot.utf8), format: .pem)
+        
+        let leaf = try NIOSSLCertificate(bytes: .init(leafCertificateForTLSIssuedFromCustomCARoot.utf8), format: .pem)
+        let leaf_privateKey = try NIOSSLPrivateKey.init(bytes: .init(privateKeyForLeafCertificate.utf8), format: .pem)
+        
+        let client_cert = try NIOSSLCertificate(bytes: .init(leafCertificateForClientAuthenticationIssuedFromCustomCARoot.utf8), format: .pem)
+        let client_privateKey = try NIOSSLPrivateKey.init(bytes: .init(privateKeyForClientAuthentication.utf8), format: .pem)
+        
+        // Client Configuration.
+        var clientConfig = TLSConfiguration.makeClientConfiguration()
+        clientConfig.renegotiationSupport = .none
+        clientConfig.certificateChain = [.certificate(client_cert)]
+        clientConfig.privateKey = .privateKey(client_privateKey)
+        
+        // Server Configuration
+        var serverConfig = TLSConfiguration.makeServerConfiguration(
+            certificateChain: [.certificate(leaf)],
+            privateKey: .privateKey(leaf_privateKey)
+        )
+        serverConfig.sendCANameList = true
+        serverConfig.trustRoots = .certificates([root])
+        serverConfig.certificateVerification = .fullVerification
+        
+        // Pulling in the logic for assertHandshakeSucceededEventLoop because there needs to be some custom handling
+        // due to the IP address and the potential for `CERTIFICATE_VERIFY_FAILED`.
+        var clientContext: NIOSSLContext!
+        var serverContext: NIOSSLContext!
+        
+        XCTAssertNoThrow(clientContext = try NIOSSLContext(configuration: clientConfig))
+        XCTAssertNoThrow(serverContext = try NIOSSLContext(configuration: serverConfig))
+
+        let group = MultiThreadedEventLoopGroup(numberOfThreads: 1)
```

Comment:
The reason I am doing this here is because I am using the contents of `assertHandshakeSucceededEventLoop` as a template to account for `NIOSSLError`'s that come through here on the client channel in `finishHandshakeForCANameTests`.

Reply:
I looked into this deeper to see if I could ditch `MultiThreadedEventLoopGroup` for the CA name unit tests; the stadard function for `assertHandshakeSucceeded` would not work so I assumed the contents of  `assertHandshakeSucceededEventLoop` two of the test functions and this did work well.